frontend/src/features/projects/assess/steps/ReviewApproveStep.tsx

```diff
@@ -46,6 +47,35 @@ export const validateTasks = (project: IProject) => {
   }, {});
 };
 
+export const DenyProjectYupSchema = Yup.object().shape({
+  publicNote: Yup.string().required('Shared note must contain a reason before denying project.'),
+});
+
+export const validateDeny = (project: IProject) => {
+  return new Promise(resolve => {
+    validateYupSchema(project, DenyProjectYupSchema).then(
+      () => resolve({}),
+      (errors: any) => resolve(yupToFormErrors(errors)),
+    );
+  });
+};
+
+export const validateApprove = (project: IProject) => {
```

Comment:
I'm not sure this will work properly. The validateYupSchema method here will run async, so the overall validateApprove method will return the yupErrors and never wait for the promise to resolve. There is a validatesync variant you could use instead I believe. Or just switch to using an await.

Reply:
It does work because of the `Promise(resolve => ...)`, however I can rewrite with an `await`